### PR TITLE
fix: convert compressed key

### PR DESCRIPTION
**Pull request by xibo1**

### CODE REVIEW

File: src/PrivateKey.cs [hunk 1]
```diff
@@ -39,6 +39,13 @@ public byte[] Sign(byte[] hash)
         {
             var hash32 = new uint256(hash);
             var signature = _bitcoinKey.SignCompact(hash32, false);
```
Comment (argfoo-web3) [inline src/PrivateKey.cs]: The if statement should be operating on the private key first before SignCompact.

if(compressed)
> OperateOnPrivateKey
SignCompact

File: src/PrivateKey.cs [hunk 2]
```diff
@@ -10,6 +10,20 @@ public class PrivateKey : IDisposable
...
 
+        private BitcoinKey NormalizedBitcoinKey
+        {
+            get
+            {
+                if (_bitcoinKey.IsCompressed)
+                {
+                    var keyBytes = _bitcoinKey.ToBytes();
+                    Array.Resize(ref keyBytes, 32);
+                    return new BitcoinKey(keyBytes, -1, false);
```
Comment (gldeng) [inline src/PrivateKey.cs]: Assign the value to _bitcoinKey

File: test/BIP39Wallet.Tests.cs [hunk 3]
```diff
@@ -105,5 +105,20 @@ public void Sign_ReturnsValidSignature()
             // Assert
             Assert.Equal(signed.ToLower(), resultSting);
         }
+
+        [Fact]
+        public void Sign_WithCompressedKeyAndDecompressedKey()
```
Comment (gldeng) [inline test/BIP39Wallet.Tests.cs]: Split the test case into two.

File: src/PrivateKey.cs (hunk 2 above)
Comment (xibo1) [reply src/PrivateKey.cs]: done

File: test/BIP39Wallet.Tests.cs (hunk 3 above)
Comment (xibo1) [reply test/BIP39Wallet.Tests.cs]: done

File: src/PrivateKey.cs [hunk 4]
```diff
@@ -7,9 +7,23 @@ namespace BIP39Wallet
...
 
+        private BitcoinKey NormalizedBitcoinKey
+        {
+            get
+            {
+                if (_bitcoinKey.IsCompressed)
+                {
+                    var keyBytes = _bitcoinKey.ToBytes();
+                    Array.Resize(ref keyBytes, 32);
+                    return _bitcoinKey = new BitcoinKey(keyBytes, -1, false);
```
Comment (gldeng) [inline src/PrivateKey.cs]:
```suggestion
                    _bitcoinKey = new BitcoinKey(keyBytes, -1, false);
```

**Review by gldeng — CHANGES_REQUESTED** [review]

**Pull request closed without merging**